### DISCUSSION TITLE
When will you support IntelliSense link path completions (# and ## triggers) in VSCode?

### DISCUSSION THREAD

**Discussion by yokaiichi** [mdx-js/mdx#2557] [Q&A] [answered]
For regular markdown files (`.md`), VSCode's IntelliSense autocomplete can support a very critical everyday authoring use case for docsite content: links that reference headers in the same file, and occasionally also links that reference a specific header in another file. It's dirt easy in VSCode to type `#` inside empty link destination delimiters `()`, and IntelliSense will display a handy list of all headers in the current `.md` file. Likewise, you can type `##` and IntelliSense will display a handy list of ALL headers in ALL `.md` files, which you can quickly filter down with typeahead.

I cannot overstate how significant this authoring shortcut can be. It's a massive time-saver when you're maintaining large doc sites. 

Unfortunately, the MDX extension for VSCode does not support this type of link path completion as of November 2024. The MDX extension supports _relative file path completion_ (using `./` or `../`), but it doesn't support link path completion (using `#` or `##`).

Do you have a roadmap item to improve your IntelliSense functionality in the MDX extension to support this type of autocomplete shortcut for `.mdx` files in VSCode? If not, I would suggest that you prioritize this usability gap.

**Comment by remcohaszing** [comment, marked as answer]
If this is supported in markdown, it should also be for MDX. If it isn’t, that’s a bug. Feel free to report it in our [issue tracker](https://github.com/mdx-js/mdx-analyzer/issues).

I work on this in my free time. I’m currently really busy and have limited time for OSS. To me personally this issue is also relatively low prio compared to some other open issues. If you want me to prioritize this, we can discuss a freelance gig. Alternatively I would more than welcome a PR that fixes it.

**Reply by yokaiichi** [reply]
Thanks. Not in a position to pay for work on this, nor am I capable of fixing this problem myself for a PR. It _is_ a blocking issue for Docusaurus adoption and authoring workflow (because it prevents us from using .mdx files in Docusaurus repos), so per your advice I've opened an issue at mdx-analyzer: [MDX Extension for VSCode doesn't support IntelliSense link path completion for # and ## triggers in .mdx files](https://github.com/mdx-js/mdx-analyzer/issues/482)
